lib/NaiveBayesTextTagger.jsm

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+this.NaiveBayesTextTagger = class NaiveBayesTextTagger {
+  constructor(model, tokenizer) {
+    this.model = model;
+    this.tokenizer = tokenizer;
+  }
+
+  /**
+   * Determines if the text belongs to class according to binary naive Bayes
+   * classifier. Returns an object containing the class label ("label"), and
+   * the log probability ("logProb") that the text belongs to that class. If
+   * the positive class is more likely, then "label" is the positive class
+   * label. If the negative class is matched, then "label" is set to null.
+   */
+  tag(text) {
+    let fv = this.tokenizer.getTfIdfVector(text, this.model.vocab_idfs);
+
+    let bestLogProb = null;
+    let bestClassId = -1;
+    let bestClassLabel = null;
+    let logSumExp = 0.0; // will be P(x). Used to create a proper probability
+    for (let classId = 0; classId < this.model.classes.length; classId++) {
```

Comment:
Nit: [for..of statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) is more succinct for array iteration. Consider replacing all the raw for loops with `for..of`?

Reply:
Did you miss this?